olive/data/component/text_generation.py

```diff
@@ -463,3 +505,12 @@ def extract_alpaca_dataset(example):
     # remove unused columns, keep only input and output
     dataset = dataset.remove_columns([col for col in dataset.column_names if col not in ["input", "output"]])
     return dataset
+
+
+def apply_template(dataset, cols, new_col, template, remove_cols=False):
+    """Apply template to column in dataset."""
+    dataset = dataset.map(
+        lambda x: {new_col: template.format(**{col: x[col] for col in cols})},
```

Comment:
Could you help give some introductions, like: the placeholder col in template must be from dataset?

Reply:
the user scenario for f-string template seems be limited.

Reply:
Like I mention in https://github.com/microsoft/Olive/pull/574#discussion_r1329091884, I am planning to support multiple different ways to set the input text. 
The f-string template is limited but it covers a lot of potential use cases while being easy to provide. for custom formatting func, the user needs to write and provide a user script, script dir, etc.